process/economics/economicsData.go

```diff
@@ -267,10 +276,18 @@ func (ed *EconomicsData) CheckValidityTxValues(tx process.TransactionWithFeeHand
 }
 
 // MaxGasLimitPerBlock will return maximum gas limit allowed per block
-func (ed *EconomicsData) MaxGasLimitPerBlock() uint64 {
+func (ed *EconomicsData) MaxGasLimitPerBlock(shardID uint32) uint64 {
+	if shardID == core.MetachainShardId {
+		return ed.maxGasLimitPerMetaBlock
+	}
 	return ed.maxGasLimitPerBlock
 }
 
+// MaxGasLimitPerMetaBlock will return maximum gas limit allowed per block
+func (ed *EconomicsData) MaxGasLimitPerMetaBlock() uint64 {
```

Comment:
why do we need this method for? `MaxGasLimitPerBlock` is enogh for the job.

Reply:
done